### DISCUSSION TITLE
Response.redirect is routing to the new page but not changing url in browser's address bar

### DISCUSSION THREAD

**Discussion by HarwinderSingh15** [vercel/next.js#65439] [Help]
### Summary

Im using nextAuth@beta version to implement authentication in my nextjs14 app. 

When user got logged in it is routing to the portal screen which is fine but after routing its not changing url in the address bar which is quite a problematic for me. I have search a lot about it but got nothing 

**PLEASE HELP**

This is my auth.config.ts file.

`
import { NextAuthConfig } from "next-auth";

export const authConfig: NextAuthConfig = {
  pages: {
    signIn: "/login",
  },
  callbacks: {
    async jwt({ token, user }) {
      if (user && user?.accessToken) {
        token.accessToken = user.accessToken;
      }
      return token;
    },
    async session({ session, user, token }) {
      session.user = {
        name: session?.user?.name,
        email: session?.user?.email,
        image: session?.user?.image,
        accessToken: token?.accessToken,
      };
      return session;
    },
    authorized({ auth, request: { nextUrl, url } }) {
        const isLoggedIn = !!auth?.user;
        const isRootOrDashboard =
          nextUrl.pathname === "/" || nextUrl.pathname.startsWith("/portal");

        if (isRootOrDashboard) {
          if (!isLoggedIn) {
            return false
          } else if (nextUrl.pathname === "/") {
            return Response.redirect(new URL("/portal", nextUrl));
          }
        } else if (isLoggedIn) {
          return Response.redirect(new URL("/portal", nextUrl));
        }

        return true;
    
    },
  },
  providers: [], // Add providers with an empty array for now
};
`

**Comment by artsztu27** [comment]
I also encountered the same problem.
Can someone help

**Comment by HarwinderSingh15** [comment]
Checkout the previous comment on this post it worked
